File: Data-Structures/Arrays/Exercises/Contains-Duplicate/Solution-2/solution2.py
import unittest

###### IMPLEMENTATION #######

class Solution:
    def containsDuplicate(self, nums: list[int]) -> bool:
        nums.sort()
        for i in range(len(nums) - 1):
            if nums[i] == nums[i+1]:
                return True

        return False

###### TESTING #######

class TestContainsDuplicate(unittest.TestCase):
    def test_1(self):
        self.assertEqual(Solution().containsDuplicate([1,2,3,1]), True)
    
    def test_2(self):
        self.assertEqual(Solution().containsDuplicate([1,2,3,4]), False)
    
    def test_3(self):
        self.assertEqual(Solution().containsDuplicate([1,1,1,3,3,4,3,2,4,2]), True)

if __name__ == '__main__':
    unittest.main()

File: Data-Structures/Arrays/Exercises/Contains-Duplicate/Solution-2-Sort/solution2.cpp
#include <iostream>
#include <vector>
#include <algorithm> 

using namespace std;

class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        int numsSize = nums.size();
         
        sort(nums.begin(), nums.end());
        for (int i = 0; i < numsSize-1; i++)
            if (nums[i] == nums[i+1])
                return true;

        return false;
    }
};

int main() {
    vector<int> nums = { 1,2,3,4 };
    string solution = Solution().containsDuplicate(nums) ? "true" : "false";
    cout << solution << "\n";

    return 0;
}